### PR TITLE
AuthZ: Improve tracing for debugging

### PR DESCRIPTION
**Pull request by gamab**
For improved debuggability, we've enhanced tracing capabilities. We realized logging wasn't the best fit for a library.

### CODE REVIEW

File: authz/multitenant_client.go [hunk 1]
```diff
@@ -181,6 +195,8 @@ func (c *LegacyClientImpl) Check(ctx context.Context, req *CheckRequest) (bool,
 		return false, nil
 	}
 
+	span.SetAttributes(attribute.String("subject", req.Caller.IDTokenClaims.Subject))
```
Comment (drclau) [inline authz/multitenant_client.go:198]: nit: if you put this line in an `else` block for the `if` above, the safety of dereferencing `IDTokenClaims` will be more explicit, plus it's slightly more future-proof.
Functionally, it's OK as it is too.

**Review by drclau — APPROVED** [review]
LGTM, w/ small optional suggestion.

**Review by aarongodin — APPROVED** [review]
LGTM, other than Claudiu's existing comment

File: authz/multitenant_client.go (hunk 1 above)
Comment (gamab) [reply authz/multitenant_client.go:198]: The existence of the IDTokenClaims subject is already checked higher up in the validate function, and the subject is used lower down without any check.

**Pull request merged**